vaadin-renderer-flow-parent/vaadin-renderer-flow-integration-tests/src/test/java/com/vaadin/flow/data/renderer/tests/ComponentRendererInNewThreadIT.java

```diff
@@ -16,25 +16,43 @@
 package com.vaadin.flow.data.renderer.tests;
 
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
 
 import com.vaadin.flow.testutil.TestPath;
 import com.vaadin.tests.AbstractComponentIT;
 
 @TestPath("vaadin-renderer-flow/component-renderer-in-new-thread")
 public class ComponentRendererInNewThreadIT extends AbstractComponentIT {
 
-    @Test
-    public void componentRendererInNewThread_uiNotNullWhileTemplateExpressionIsCalculated() {
+    @Before
+    public void init() {
         open();
-        findElement(By.id("add-component")).click();
+    }
+
+    @Test
+    public void addComponentRendererBeforeAttach_componentIsRendered() {
```

Comment:
**note:** This test is a modern version of `componentRendererInNewThread_uiNotNullWhileTemplateExpressionIsCalculated` which covered https://github.com/vaadin/flow-components/pull/5534